students/BrianE/session08/test_circle.py

```diff
@@ -0,0 +1,197 @@
+from circle import Circle, Sphere
+import pytest
+import math
+
+
+def test_circle_init():
+    """ Verify circle init functions properly """
+    a_circle = Circle(5)
+    assert isinstance(a_circle, Circle)
+    assert a_circle.radius == 5
+
+
+def test_circle_diameter():
+    """ Verify circle diameter is set to 2x radius """
+    a_circle = Circle(5)
+    assert a_circle.diameter == 5 * 2
+
+
+def test_circle_diameter_radius_sync():
+    """ Verify updating diameter and radius maintain sync when updating either value """
+    a_circle = Circle(5)
+
+    # changing diameter results in updated radius
+    a_circle.diameter = 20
+    assert a_circle.radius == 10
+
+    # changing radius results in updated diameter
+    a_circle.radius = 7
+    assert a_circle.diameter == 14
+
+
+def test_circle_area_calculation():
+    """ Verify circle area calculation """
+    a_circle = Circle(5)
+    assert a_circle.area == math.pi * 5 ** 2
+
+
+def test_circle_area_set():
+    """ Verify circle area cannot be set by user """
+    a_circle = Circle(5)
+    with pytest.raises(AttributeError):
+        a_circle.area = 25
+
+
+def test_circle_str():
+    """ Verify circle __str__ output """
+    radius = 5
+    a_circle = Circle(radius)
+    assert a_circle.__str__() == f"Circle with radius: {radius}"
+
+
+def test_circle_repr():
+    """ Verify circle __repr__ output """
+    a_circle = Circle(5)
+    assert a_circle.__repr__() == f"Circle(5)"
+
+
+def test_circle_add():
+    """ Verify adding two circles together functions properly """
+    a_circle = Circle(5)
+    b_circle = Circle(8)
+    c_circle = a_circle + b_circle
+    assert isinstance(c_circle, Circle)
```

Comment:
Nice check here! You can also create the new Circle and compare against that
assert c_circle == Circle(13)